app/tasks/maintenance.php

```diff
@@ -54,6 +97,29 @@ function notifyDeleteConnections()
             ]);
         }
 
+        function renewCertificates($dbForConsole)
+        {
+            $time = date('d-m-Y H:i:s', time());
+            /** @var Utopia\Database\Database $dbForConsole */
+
+            $certificates = $dbForConsole->find('certificates', [
+                new Query('attempts', Query::TYPE_LESSER, [5]), // Maximum 5 attempts
+                new Query('renewDate', Query::TYPE_LESSEREQUAL, [\time()]) // includes 60 days cooldown (we have 30 days to renew)
+            ], 300); // Limit 300 comes from LetsEncrypt (orders per 3 hours)
```

Comment:
What do you mean by `orders per 3 hours`?

Reply:
LetsEncrypt has a limit of 300 orders per 3 hours. The order I believe means successful certificate generation for a domain that does need a regeneration. The limit is different for the domain that doesn't need it, but that's none of our concerns.

If we attempt to renew more than 300 domains (that we know actually need renewal), we would be hitting API limit anyway. Limit 300 makes sense to me, or even, maybe set it to just 200. That way other work can be also done within the maintenance hour. If we renew use up all 300 orders, if user adds new custom domain, it would always fail due to API limit.